### ISSUE TITLE
Bundle path-to-regexp helper

### ISSUE THREAD

**Issue by daun** [swup/swup#662]
**Describe the problem**

Three official plugins are now making use of the [path-to-regexp](https://www.npmjs.com/package/path-to-regexp) library for matching URL patterns (including the work-in-progress fragment plugin). This could lead to duplicate versions being bundled by consumers of these plugins.

**Describe the proposed solution**

Require `path-to-regexp` from swup core, create a helper around it with a stable API, and offer it as a named export that plugins can consume. Example usage:

```js
import { matchPath } from 'swup';

export default class SomePlugin {
  constuctor({ routes }) {
    this.patterns = routes.map(route => matchPath(route));
  }
}
```

This is identical to how we've dealt with `delegate-it` for event delegation. It won't be bundled into the output, but will be available as a named export.

**Alternatives considered**

- Leave this to the plugins themselves, risking duplicate versions

**How important is this feature to you?**

- [x] Nice to have
- [ ] Would make my life a lot easier
- [ ] I cannot use swup without it

**Comment by hirasso** [comment]
Shouldn't we rather just re-export the library without a compatibility layer?

**Comment by daun** [comment]
Our recent experience with `delegate-it` kind of convinced me to always add a minimal wrapper around everything. `delegate-it` had three consecutive breaking API changes within less than a year. Having a wrapper with a stable API around it has really helped in allowing us to update it behind the scenes with propagating the breaking change — the re-exported custom API remains stable for plugins.

**Comment by daun** [comment]
Implemented in #694
